lib/ansible/plugins/action/__init__.py

```diff
@@ -549,7 +563,7 @@ def _execute_module(self, module_name=None, module_args=None, tmp=None, task_var
 
         # Fix permissions of the tmp path and tmp files.  This should be
         # called after all files have been transferred.
-        self._fixup_perms(tmp, remote_user, recursive=True)
+        self._fixup_perms(tmp, remote_user, recursive=True, execute=is_binary)
```

Comment:
The two things I was thinking about were: When someone sets ANSIBLE_KEEP_REMOTE_FILES=1 and then wants to execute the wrapper and async.  async currently sets execute=True so that async_wrapper can execute the module that was copied over.  Started me wondering if there's any harm to just setting execute=True by default.